CONTRIBUTING.md

```diff
@@ -6,13 +6,75 @@ Microsoft Authentication Library (MSAL) for .NET welcomes new contributors.  Thi
+
+### Identifying scale
+
+If you would like to contribute to one of our repositories, first identify the scale of what you would like to change. If it is small (grammar/spelling or a bug fix), feel free to start working on a fix. If you are submitting a feature or a substantial code contribution, please discuss it with the team and ensure it follows the product roadmap. You might also read these two blogs posts on contributing code: [Open Source Contribution Etiquette by Miguel de Icaza](http://tirania.org/blog/archive/2010/Dec-31.html) and [Don't "Push" Your Pull Requests by Ilya Grigorik](https://www.igvita.com/2011/12/19/dont-push-your-pull-requests/). All code submissions will be rigorously reviewed and tested further by the team, and only those that meet an extremely high bar for both quality and design/roadmap appropriateness will be merged into the source.
+
+### Before submitting a pull request
+
+Before submitting a pull request, make sure that it passes the following requirements:
+
+- You found an existing issue with a `help-wanted` label or discussed with the team and agreed on adding a new issue with that label.
+- You posted a high-level description of how the changes will be implemented and received a positive acknowledgement from the team before getting too committed to the approach or investing too much effort in implementing it.
+- You added test coverage following existing patterns within the codebase.
+- Your code matches the existing syntax conventions within the codebase.
+- Your PR is small, focused, and avoids making unrelated changes.
+
+If your pull request contains any of the below, it's less likely to be merged.
+
+- Changes that break backward compatibility
+- Changes that are only wanted by one person or company. Changes need to benefit a large enough proportion of developers using our auth libraries.
+- Changes that add entirely new feature areas without prior agreement
+- Changes that are mostly about refactoring existing code or code style
+
+Very large PRs that would take hours to review (remember, we're trying to help lots of people at once). For larger work areas, please discuss with us to find ways of breaking it down into smaller, incremental pieces that can go into separate PRs.
+
+### During a pull request review
+
+A core contributor will review your pull request and provide feedback. To ensure that there is not a large backlog of inactive PRs, the pull request will be marked as stale after two weeks of no activity. After another four days, it will be closed.
```

Comment:
Has this automation been completed in other repos?

Reply:
No, this has not and I don't think we plan to.

Reply:
I've merged this into the Merging a pull request section.